packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -335,25 +342,28 @@ export class Key extends KeyBase {
   }
 
   /**
-   * Let users from this account admin this key.
+   * Let users or IAM policies from this account admin this key.
+   * @link https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
    * @link https://aws.amazon.com/premiumsupport/knowledge-center/update-key-policy-future/
    */
-  private allowAccountToAdmin() {
-    const actions = [
-      "kms:Create*",
-      "kms:Describe*",
-      "kms:Enable*",
-      "kms:List*",
-      "kms:Put*",
-      "kms:Update*",
-      "kms:Revoke*",
-      "kms:Disable*",
-      "kms:Get*",
-      "kms:Delete*",
-      "kms:ScheduleKeyDeletion",
-      "kms:CancelKeyDeletion",
-      "kms:GenerateDataKey"
-    ];
+  private allowAccountToAdmin(enablePolicyControl: boolean) {
+    const actions = enablePolicyControl ?
+      ['kms:*'] :
+      [
+        "kms:Create*",
```

Comment:
these permissions are un-altered just moved.

Reply:
It's interesting tagging and untagging permissions are not granted. This means that if you add or remove after the initial stack it will warn the user but not implement the change.